### DISCUSSION TITLE
When to use Neuralforecast and Statsforecast?

### DISCUSSION THREAD

**Discussion by rllyryan** [Nixtla/neuralforecast#362] [Q&A] [answered]
Dear to whom it may concern,

I would like to find out in what kind of scenarios is (Neuralforecast or Statsforecast) more suitable for? I do not have a solid background in the field of forecasting algorithms but I understand the gist of simple forecasting methods like ARIMA and its variants.

I believe this question is similar to the comparison of ARIMA VS Neural Networks.

If someone has a brief summary of the benefits and use cases of these models (available in the libraries), please do share them!

Thank you in advance!

**Comment by kdgutier** [comment, marked as answer]
Hi @rllyryan,

Thanks for such an interesting question; here are some neural forecasting advantages:

1. **Performance Gains on Large datasets**: Statistical models work well when the series are stationary and 'predictable'. Most statistical methods operate on single series. In contrast, modern neural forecasting methods are highly flexible models that estimate their parameters from a large set of correlated series (cross-learning). The deep learning performance gains over statistical methods are evident with bigger datasets (see M4 competition).

2. **Cold-start Predictions**: When trained as a global model, they allow knowledge to be transferred from other similar series, enabling one of the most exciting neural forecasting method's abilities: provide accurate predictions for series with little to no prior history.

3. **Handling Heterogeneous seasonalities and exogenous effects**: In the presence of heterogeneous seasonalities or exogenous effects, it is extremely hard to maintain a complex forecasting system based on tailored statistical models for each series. Unified models greatly simplify forecasting pipelines.

4. **Automated Feature Engineering**: Neural networks offer the ability to take raw input data and learn useful representations, alleviating the efforts of manual feature engineering, this ability double-downs on the neural networks as a solution for a unified forecasting system.

5. **Task-specific Optimization**: Statistical methods learn a distribution under strong assumptions that may only sometimes hold. Neural forecasting can directly target metrics with business significance. For example, you can optimize a network's parameters with the quantile loss that differentiates under and over-prediction that may have different implications in inventory management.

Neural Forecasting is an active research field, with plenty of interesting challenges:

1. **Data hungry methods**: Cross-learning on large datasets enables deep learning to improve forecasting accuracy over statistical methods. Designing methods that overcome this necessity for large datasets will expand their usability and impact.

2. **Limited Interpretability**: Our understanding of the model's predictions can be as crucial as their accuracy in several situations. It creates trust in its predictions and improves our knowledge of the modeled processes. The non-interpretability of vanilla networks is a barrier to adopting neural forecasting.

3. **Computational Cost**: Neural network's optimization is computationally expensive. The costs often require specialized hardware like GPUs. Programming expertise and computational costs can be an obstacle for newcomers.

4. **Usability**: Finally, despite their pipeline simplification capabilities, networks are complex models with many hyperparameter decisions for the user. It is common for users to need more knowledge of the intricate inner workings of the models, which makes realizing neural forecasting potential difficult. The challenge here is to simplify their usage and give users enough tools and documentation to improve their usability.

Let me know your thoughts :)
Hope this helps.

**Reply by rllyryan** [reply]
Oh wow, thank you @kdgutier for the detailed explanation of the benefits!!! Hope to implement this is real world scenarios. I am actually tackling an inventory management problem for warehouses, and hope that maybe this library could be the answer to it!

**Could I also ask whether there is a benchmarking done to compare the two libraries of Statsforecast and Neuralforecast?**

**Reply by kdgutier** [reply]
We are working on that.
